src/h3lib/lib/geoCoord.c

```diff
@@ -332,3 +332,74 @@ int64_t H3_EXPORT(numHexagons)(int res) {
                                    569707381193162L};
     return nums[res];
 }
+
+// do we have a preference towards passing pointers instead of the struct?
+// duplicate of _geoDistRads?
+double haversine_points(GeoCoord a, GeoCoord b) {
+    // Haversine distance between two points.
+    // Input/output: both in radians
+
+    double x, y, z;
+
+    a.lon -= b.lon;
+
+    x = cos(a.lon) * cos(a.lat) - cos(b.lat);
+    y = sin(a.lon) * cos(a.lat);
+    z = sin(a.lat) - sin(b.lat);
+
+    return 2 * asin(0.5 * sqrt(x * x + y * y + z * z));
+}
```

Comment:
Looks like this helper function may be a duplicate of `_geoDistRads`.
However, the math between them is very different. Not sure which one we should prefer.

@isaacbrodsky, I actually got this math from an example you have elsewhere in the codebase. Do you have a reference for it? I can't seem to find this version anywhere else on the web...